### ISSUE TITLE
sql: DROP SEQUENCE CASCADE, DROP TABLE CASCADE => sequence

### ISSUE THREAD

**Issue by vilterp** [cockroachdb/cockroach#20965]
Expected:

- `DROP SEQUENCE seq CASCADE` drops the associated table
- `DROP TABLE tbl CASCADE`, when `tbl` uses a sequence, drops the sequence

Current state:

- `DROP SEQUENCE CASCADE` doesn't parse
- `DROP TABLE CASCADE` doesn't affect sequences

**Comment by bdarnell** [comment]
Is this the same behavior as postgres? According to the docs, DROP TABLE tbl CASCADE drops objects that depend on the table, not the other way around.

**Comment by vilterp** [comment]
My mistake; edited issue to match PG behavior.

**Comment by arulajmani** [comment]
Updating the state of this issue right now: 

- DROP SEQUENCE CASCADE does parse, but the dropBehavior is ignored and it simply behaves like RESTRICT/no option given.
- A comment on the Postgres behavior, a sequence used by the table (and no other table) is only dropped if a column belonging to the table being dropped owns the sequence. Once #26382 is closed and we have the concept of ownership in Cockroach, dropping a table that owns a sequence will be supported.
- The only extra work required after #26382 goes in will be to support DROP SEQUENCE CASCADE. The only blocker on this front is clearing out the default operation for columns that are depended on by the sequence being dropped. Note that in pg, only the default operation is affected when doing a CASCADE -- the entire table is not dropped (the docs are a bit misleading on this). 
- Dropping a table that has a column that owns a sequence, and that sequence is used by other columns should follow the same dropBehavior behavior as supplied to by the drop table command. Currently, it will follow RESTRICT regardless, as CASCADE for sequences does not exist.

tldr: Once #26382 goes in, I think the only work required to close this issue would be the last two points.

**Comment by the-ericwang35** [comment]
This should be resolved by https://github.com/cockroachdb/cockroach/pull/60744
